File: examples/beginner-demo/v1/service/order_service.py
"""服务层：订单业务。"""

from domain.order import Order
from repository.order_repository import OrderRepository


class OrderService:
    def __init__(self, repo: OrderRepository):
        self.repo = repo

    def create_order(self, order_id: str, amount: float) -> Order:
        order = Order(order_id, amount)
        self.repo.save(order)
        return order

    def pay(self, order_id: str) -> Order:
        order = self.repo.find(order_id)
        order.mark_paid()
        self.repo.save(order)
        return order

File: eval/fixtures/java-order/src/main/java/com/example/service/OrderService.java
package com.example.service;

import com.example.model.Order;
import com.example.repository.OrderRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional
public class OrderService {

    private final OrderRepository orderRepository;

    public OrderService(OrderRepository orderRepository) {
        this.orderRepository = orderRepository;
    }

    public Order createOrder(Order order) {
        return orderRepository.save(order);
    }

    public Order getOrder(Long id) {
        return orderRepository.findById(id).orElse(null);
    }
}
